pkg/rust/cargo/parse.go

```diff
@@ -27,18 +31,57 @@ func NewParser() types.Parser {
 
 func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
 	var lockfile Lockfile
-	if _, err := toml.DecodeReader(r, &lockfile); err != nil {
+	decoder := toml.NewDecoder(r)
+	if _, err := decoder.Decode(&lockfile); err != nil {
 		return nil, nil, xerrors.Errorf("decode error: %w", err)
 	}
 
+	if _, err := r.Seek(0, io.SeekStart); err != nil {
+		return nil, nil, xerrors.Errorf("seek error: %w", err)
+	}
+
+	// naive parser to get line numbers by package from lock file
+	pkgParser := naivePkgParser{r: r}
+	lineNumIdx := pkgParser.parse()
+
 	var libs []types.Library
+	var deps []types.Dependency
 	for _, pkg := range lockfile.Packages {
 		pkgID := utils.PackageID(pkg.Name, pkg.Version)
-		libs = append(libs, types.Library{
+		lib := types.Library{
 			ID:      pkgID,
 			Name:    pkg.Name,
 			Version: pkg.Version,
-		})
+		}
+		if pos, ok := lineNumIdx[pkgID]; ok {
+			lib.Locations = []types.Location{{StartLine: pos.start, EndLine: pos.end}}
+		}
+
+		libs = append(libs, lib)
+		dep := parseDependencies(pkgID, pkg)
+		if dep != nil {
+			deps = append(deps, *dep)
+		}
+	}
+	return libs, deps, nil
+}
+func parseDependencies(pkgId string, pkg cargoPkg) *types.Dependency {
+	dependOn := []string{}
+	for _, pkgDep := range pkg.Dependencies {
+		fields := strings.Fields(pkgDep)
```

Comment:
Can you add comment about this field schema?
if i understand correctly it is: `<pkg_name> <pkg_version> (registry+<link>)` , right?